src/__tests__/e2e.test.js

```diff
@@ -585,6 +587,51 @@ describe('e2e', () => {
     });
   });
 
+  it('does not cause unnecessary updates due to context changes', () => {
+    let innerRenderCount = 0;
+
+    const OuterComponent = track()(props => props.children);
+
+    const MiddleComponent = track()(
+      React.memo(
+        props => props.children,
+        (props, prevProps) => props.middleProp === prevProps.middleProp
+      )
+    );
+
+    const InnerComponent = track()(() => {
+      innerRenderCount += 1;
+      return null;
+    });
+
+    const App = track()(() => {
+      const [props, setProps] = React.useState({});
```

Comment:
I know these can be called anything you want, but by [convention/docs](https://reactjs.org/docs/hooks-reference.html#usestate) these are usually:

```js
const [state, setState] = React.useState({});
```